pkg/handler/v2/external_cluster/eks.go

```diff
@@ -223,16 +223,44 @@ func patchEKSMD(old, new *apiv2.ExternalClusterMachineDeployment, secretKeySelec
 	clusterName := cloudSpec.EKS.Name
 	nodeGroupName := new.NodeDeployment.Name
 
+	currentReplicas := old.NodeDeployment.Spec.Replicas
 	desiredReplicas := new.NodeDeployment.Spec.Replicas
-	_, err = resizeEKSNodeGroup(client, clusterName, nodeGroupName, int64(desiredReplicas))
+	if desiredReplicas != currentReplicas {
+		_, err = resizeEKSNodeGroup(client, clusterName, nodeGroupName, int64(currentReplicas), int64(desiredReplicas))
+		if err != nil {
+			return nil, err
+		}
+		new.NodeDeployment.Status.Replicas = desiredReplicas
```

Comment:
for GKE when I update the version then upgrading NodeGroup fails because of pending operation. If you have the same situation then the endpoint will always throw the error even the size will be changed. Please check this. Mybe 'return' will be needed in this block